mautic/mautic
# Campaign infinite loop due to restarts validation

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | 



#### Description:

Our customers sometimes configure a campaign that restart itself without any delay. This is wasting our resources as the contacts are flowing through the campaign in an infinite loop as fast as they can. It generates a lot of data as well which slows down the campaign execution. We've created this validation to slow down the loops to minimally once/30 minute. We don't see any reason why a contact should loop through the campaign any faster.

<img width="619" alt="Screenshot 2024-06-12 at 18 02 47" src="https://github.com/acquia/mc-cs/assets/1235442/467128c5-8169-48bb-ad4f-f73b3b13e80b">
<img width="617" alt="Screenshot 2024-06-12 at 18 02 27" src="https://github.com/acquia/mc-cs/assets/1235442/ce25cac6-60bd-49fa-8a13-8d34ceedf816">
<img width="640" alt="Screenshot 2024-06-12 at 18 06 31" src="https://github.com/acquia/mc-cs/assets/1235442/b428eb2d-001c-4f78-9ca4-b9d789162901">


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new campaign with the "Change campaign membership" action.
3. Try to save it without any delay. It should tell you that you have to add a delay.
4. Try to save it with a delay with less than 30 minutes. It should tell you it must be at least 30 minutes.
5. Try to save it with a delay greater than 30 minutes. It should save the campaign event.

#### Other areas of Mautic that may be affected by the change:
1. None

#### List deprecations along with the new alternative:
1. None



[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. The validation covers all the variations of saving this event via UI and API
